test/integration/gsp-extension/test/index.test.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-env jest */
+
+import fs from 'fs-extra'
+import { join } from 'path'
+import { findPort, killApp, nextBuild, nextStart } from 'next-test-utils'
+
+jest.setTimeout(1000 * 60 * 2)
+const appDir = join(__dirname, '..')
+
+let appPort
+let app
+
+describe('GS(S)P with file extension', () => {
+  beforeAll(async () => {
+    await fs.remove(join(appDir, '.next'))
+    const { code } = await nextBuild(appDir)
+    if (code !== 0) throw new Error(`build failed with code ${code}`)
+
+    appPort = await findPort()
+    app = await nextStart(appDir, appPort)
+  })
+  afterAll(() => killApp(app))
+
+  it('should support slug with different extensions', async () => {
```

Comment:
Can we add additional checks to ensure paths with these names revalidate and return content correctly for both the `_next/data` variant and direct visiting? 

Also would the `_next/data` path be expected to include the extension in this case since it was returned from `getStaticPaths`?